test/sql/catalog/function/test_select_macro.test

```diff
@@ -0,0 +1,57 @@
+# name: test/sql/catalog/function/test_select_macro.test
+# description: Test Select Macro
+# group: [function]
+
+statement ok
+PRAGMA enable_verification
+
+
+statement ok
+CREATE TABLE test_tbl (id INT, name string);
+
+statement ok
+CREATE TABLE test2_tbl (id INT, name string);
+
+statement ok
+CREATE TABLE greek_tbl (id INT, name string);
+
+statement ok
+INSERT INTO test_tbl values (1,'tom'), (2,'dick'),(3,'harry'), (4,'mary'), (5,'mungo'), (6,'midge');  
+
+statement ok
+INSERT INTO test_tbl values (20,'andrew'), (21,'boris'),(22,'Caleb'), (23,'david'), (24,'evan');
+
+statement ok
+INSERT INTO  greek_tbl values (1, 'alpha'), (2, 'beta'), (3, 'gamma'), (4, 'delta'), (5, 'epsilon'),(6, 'zeta'), (7, 'eta') , (8, 'theta'), (9, 'iota') , (10, 'kappa'); 
+
+statement ok
+CREATE MACRO TABLE xt(a,_name) as select * from test_tbl where(id>=a or name=_name);
+
+statement ok
+CREATE MACRO TABLE xt2(a,_name) as select * from test_tbl where(id>=a or name like _name);
+
+statement ok
+CREATE MACRO TABLE sgreek(a,b,c) as select a,b from greek_tbl where(id >= c);
+
+query II
+( SELECT FUNCTION xt(1, 'tom') UNION SELECT FUNCTION  xt2(1, '%%%') ) INTERSECT SELECT FUNCTION xt(100,'midge');
+----
+6	midge
+
+query II
+(SELECT FUNCTION xt(1, 'tom') EXCEPT SELECT FUNCTION xt(20,'tom' )) INTERSECT SELECT FUNCTION xt(100,'harry');
+----
+3	harry
+
+query II
+SELECT	FUNCTION  xt(200,'andrew');
+----
+20	andrew
+
+query II
+SELECT FUNCTION xt2(100,'m%');
```

Comment:
Should this syntax not be `SELECT * FROM xt2(100, 'm%');`

Reply:
OK I wil go ahead and implement your suggested syntax.